### PR TITLE
History page

### PR DESCRIPTION
**Pull request by larazayec**

### CODE REVIEW

File: Test2/Pages/HistoryPage.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+namespace Test2.Pages
+{
+    internal class HistoryPage 
+    {
+        private IWebDriver driver;
+        public HistoryPage(IWebDriver webDriver)
+        {
+            driver = webDriver;
+        }
+        public List<IWebElement> LastResult => driver.FindElements(By.XPath("//table/tr[@class='data-td'][1]/td")).ToList();
```
Comment (varajan) [inline Test2/Pages/HistoryPage.cs]: Please add a blank line and make it `private`
```suggestion
        }
        
        privatge List<IWebElement> LastResult => driver.FindElements(By.XPath("//table/tr[@class='data-td'][1]/td")).ToList();
```

File: Test2/Pages/HistoryPage.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        public List<IWebElement> Rows => driver.FindElements(By.XPath("//table/tr[@class='data-td']")).ToList();
+        public IWebElement ClearButton => driver.FindElement(By.Id("clear"));
+        public IWebElement HistoryButton => driver.FindElement(By.XPath("//div[@class='history link btn btn-link']"));
+        public void Open()
```
Comment (varajan) [inline Test2/Pages/HistoryPage.cs]: Please add a blank line.
```suggestion
        public IWebElement HistoryButton => driver.FindElement(By.XPath("//div[@class='history link btn btn-link']"));

                public void Open()
```

File: Test2/Pages/HistoryPage.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            {
+                List<string> actuale = new List<string>();
+                foreach (IWebElement element in LastResult)
+                {
+                    actuale.Add(element.Text);
+                }
+                
+                return actuale;
+            }
+        }
```
Comment (varajan) [inline Test2/Pages/HistoryPage.cs]: As this is 'the first table line', so you can make it shorter:
```suggestion
        public List<string> LastResults => TableValues[1];
```

File: Test2/Pages/HistoryPage.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+{
+    internal class HistoryPage 
+    {
+        private IWebDriver driver;
+        public HistoryPage(IWebDriver webDriver)
+        {
+            driver = webDriver;
+        }
+        public List<IWebElement> LastResult => driver.FindElements(By.XPath("//table/tr[@class='data-td'][1]/td")).ToList();
+        public List<IWebElement> Rows => driver.FindElements(By.XPath("//table/tr[@class='data-td']")).ToList();
```
Comment (varajan) [inline Test2/Pages/HistoryPage.cs]: You use this 'Rows' only to get history size, so you can make it shorter and re-use the 'TableValues' property:
```suggestion
        public int HistoryCount => TableValues.Count() - 1;
```

File: Test2/Tests/HistoryPageTests.cs [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    internal class HistoryPageTests : BaseTest
+    {
+        [SetUp]
+        public void SetUp()
+        {
+            OpenDriver();
+            SettingPage settingPage = new SettingPage(driver);
+            settingPage.Open();
+            settingPage.SaveNumberFormat("123,456,789.00");
+            driver.Url = "https://localhost:5001/History";
```
Comment (varajan) [inline Test2/Tests/HistoryPageTests.cs]: You don't need this line here. You always open the History page in tests.
```suggestion
```

File: Test2/Tests/HistoryPageTests.cs [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+                List<IWebElement> cells = row.FindElements(By.XPath(".//td")).ToList();
+
+                foreach (IWebElement cell in cells)
+                {
+                    RowValues.Add(cell.Text);
+                }
+
+                tabelValues.Add(RowValues);
+            }
+        }
```
Comment (varajan) [inline Test2/Tests/HistoryPageTests.cs]: Please delete this test.

File: Test2/Tests/HistoryPageTests.cs [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+                List<string> rowValues = new List<string>();
+                for (int colNumber = 0; colNumber < 8; colNumber++)
+                {
+                    IWebElement cell = driver.FindElement(By.XPath($"//tr[{rowNumber+1}]//*[{colNumber+1}]"));
+                    string cellText = cell.Text;
+                    rowValues.Add(cellText);
+                }
+                tabelValues.Add(rowValues);
+            }
+        }
```
Comment (varajan) [inline Test2/Tests/HistoryPageTests.cs]: And this.

File: Test2/Tests/HistoryPageTests.cs [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+
+        [Test]
+        public void CheckElements()
```
Comment (varajan) [inline Test2/Tests/HistoryPageTests.cs]: Please rename this test.

File: Test2/Tests/HistoryPageTests.cs [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+        {
+            new HistoryPage(driver).Open();
+            HistoryPage historyPage = new HistoryPage(driver);
+            CalculatorPage calculatorPage = new CalculatorPage(driver);
+            List<List<string>> expected2 = new List<List<string>>() {
```
Comment (varajan) [inline Test2/Tests/HistoryPageTests.cs]: Please rename this variable and add a line-break before `{`
```suggestion
            List<List<string>> expected = new List<List<string>>()
            {
```

File: Test2/Tests/HistoryPageTests.cs [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+            for (int i = 0; i < 5; i++)
+            {
+                calculatorPage.Open();
+                calculatorPage.Calculate("200", "100", "365", "01", "January", "2022");
+            }
+            for (int i = 0; i < 5; i++)
+            {
+                calculatorPage.Open();
+                calculatorPage.Calculate("1000", "50", "360", "01", "January", "2022");
+            }
```
Comment (varajan) [inline Test2/Tests/HistoryPageTests.cs]: You can open the `calculatorPage` only once:

```suggestion
            historyPage.ClearButton.Click();
            calculatorPage.Open();
            for (int i = 0; i < 5; i++)
            {
                calculatorPage.Calculate("200", "100", "365", "01", "January", "2022");
            }
            for (int i = 0; i < 5; i++)
            {
                calculatorPage.Calculate("1000", "50", "360", "01", "January", "2022");
            }
```

**Review by Varajanko — APPROVED** [review]

**Review by Varajanko — APPROVED** [review]

**Review by Varajanko — APPROVED** [review]
Approved

**Review by Varajanko — APPROVED** [review]
Ok

**Review by varajan — APPROVED** [review]

**Pull request merged**